packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -2012,6 +1993,50 @@ fn new_js_string<'s>(
     v8::String::new(scope, s).expect(&format!("Could not create JS String: {s}"))
 }
 
+/// Helper function to get the exception from a [`v8::TryCatch`]
+///
+/// # Panics
+///
+/// Panics if the try catch scope didn't catch any exception.
+fn exception_as_error<'s, 'p: 's, S>(try_catch_scope: &mut v8::TryCatch<'s, S>) -> Error
+where
+    v8::TryCatch<'s, S>: AsMut<v8::HandleScope<'p, ()>>,
+    v8::TryCatch<'s, S>: AsMut<v8::HandleScope<'p, v8::Context>>,
+{
+    let exception = try_catch_scope
+        .exception()
+        .expect("Expected try catch scope to have caught an exception");
+    let exception_message = try_catch_scope.message();
+
+    let mut returned_string = exception
+        .to_string(try_catch_scope.as_mut())
+        .unwrap()
+        .to_rust_string_lossy(try_catch_scope.as_mut());
+
+    if let Some(exception_message) = exception_message {
+        returned_string.push_str(", Exception message: ");
```

Comment:
Instead of appending to the exception title, we might want to add another Error variant (maybe `V8Exception(exception, Option<message>)`?).

Reply:
I named it `JavascriptException` https://github.com/hashintel/hash/pull/493/commits/6440a6a3a3e547a0493b724a9a9042434d75011e